### PR TITLE
Validate TargetSamplesPerSecond is positive

### PR DESCRIPTION
**Pull request by Atul-Koundal**
## Problem

`newPostAggregator` validates `CalculationInterval`, `AggregationBuckets`, and `BucketsForCalculation`, but not `TargetSamplesPerSecond`. A configured value of `0` is accepted, and `withinTolerance` then divides by zero:

```
withinTolerance(0, 0) -> NaN
withinTolerance(5, 0) -> +Inf
```

Both compare false against `DeltaTolerance`, so the short-circuit in `calculateProbability` never fires and every operation's probability is recalculated on every interval instead of only when it drifts outside tolerance.

## Changes

- Reject TargetSamplesPerSecond values that are non-positive or non-finite (NaN, ±Inf) in newPostAggregator. All three slip past the tolerance check: NaN and +Inf both compare false against DeltaTolerance, and 0 divides by zero.
- Tighten the existing `CalculationInterval` / `AggregationBuckets` check from `== 0` to `<= 0`. A negative `CalculationInterval` panics in `time.NewTicker` (`panic: non-positive interval for NewTicker`) and a negative `AggregationBuckets` panics on the slice in `prependThroughputBucket`. Raised by Copilot in review.
- Update six test fixtures that relied on the previously unvalidated zero, and add test cases for the negative values.

## Testing

Before this change, a config with `target_samples_per_second: 0` and the `adaptive_sampling` processor in the pipeline started normally. After:

```
Error: cannot start pipelines: failed to start "adaptive_sampling" processor: failed to create the adaptive sampling aggregator: TargetSamplesPerSecond must be greater than 0
```

Scope note: this is only reached via the `adaptive_sampling` processor. A config with only the `remote_sampling` extension does not construct a `PostAggregator` and still starts with `0`.

`make fmt`, `make lint` (0 issues), and `go test ./internal/... ./cmd/...` pass.

---

I chose to error rather than default the value, matching how the sibling options are handled in the same constructor. Defaulting to `DefaultOptions().TargetSamplesPerSecond` would avoid the fixture churn, happy to switch if that's preferred.

### CODE REVIEW

File: internal/sampling/samplingstrategy/adaptive/post_aggregator.go [hunk 1]
```diff
@@ -106,6 +107,9 @@ func newPostAggregator(
 	if opts.BucketsForCalculation < 1 {
 		return nil, errBucketsForCalculation
 	}
```
[the remark this answers is not in the document]
Comment (Atul-Koundal) [reply internal/sampling/samplingstrategy/adaptive/post_aggregator.go:109]: Good catch, confirmed both. `time.NewTicker` panics with `non-positive interval for NewTicker` on a non-positive `CalculationInterval`, and a negative `AggregationBuckets` panics on the slice at `post_aggregator.go:205`.

Changed the existing check to `<= 0` for both and added test cases for the negative values.

One note: `provider.go:98` also calls `NewTicker`, but on `followerRefreshInterval`, which is a hardcoded constant rather than user config so that path isn't reachable from configuration.

File: internal/sampling/samplingstrategy/adaptive/post_aggregator.go [hunk 2]
```diff
@@ -100,12 +101,15 @@ func newPostAggregator(
...
 	logger *zap.Logger,
 ) (*PostAggregator, error) {
-	if opts.CalculationInterval == 0 || opts.AggregationBuckets == 0 {
+	if opts.CalculationInterval <= 0 || opts.AggregationBuckets <= 0 {
 		return nil, errNonZero
 	}
 	if opts.BucketsForCalculation < 1 {
 		return nil, errBucketsForCalculation
 	}
+	if opts.TargetSamplesPerSecond <= 0 {
```
[the remark this answers is not in the document]
Comment (Atul-Koundal) [reply internal/sampling/samplingstrategy/adaptive/post_aggregator.go]: Error: cannot start pipelines: failed to start "adaptive_sampling" processor: failed to create the adaptive sampling aggregator: TargetSamplesPerSecond must be a finite number greater than 0

[the remark this answers is not in the document]
Comment (Atul-Koundal) [reply internal/sampling/samplingstrategy/adaptive/post_aggregator.go]: Addressed, added math.IsNaN and math.IsInf checks with regression cases for both, and updated the error message to "must be a finite number greater than 0". Verified that a config with target_samples_per_second: .nan or .inf started clean before this change and now fails at startup.

**Comment by jkowall** [comment]
@Atul-Koundal Thanks for addressing the Copilot NaN/Inf feedback — that looks good on the current head.

This is currently conflicting with `main` (DIRTY). Please rebase onto current main, get required CI green, then ping me and I'll do a final review/approve.

**Comment by Atul-Koundal** [comment]
@jkowall Thanks for reviewing! Rebased onto the latest `main` and ready for the final review.

**Review by jkowall — APPROVED** [review]
LGTM. Fail-closed on non-finite/non-positive TargetSamplesPerSecond is the right contract, and the <=0 tightening for CalculationInterval/AggregationBuckets is good.

Optional nit (non-blocking): in TestConstructorFailure, set AggregationBuckets=1 before the CalculationInterval=-1 case so that path is isolated.

**Comment by jkowall** [comment]
@Atul-Koundal CI is red on `lint-fmt` — the new `TargetSamplesPerSecond` lines in `aggregator_test.go` aren't gofmt-aligned. Please run `make fmt`, push, and ping me again.

**Comment by Atul-Koundal** [comment]
@jkowall Ran `make fmt` and pushed the formatting fix. Ready for the final review once CI is green. Thanks!

**Review by yurishkuro — APPROVED** [review]

**Comment by yurishkuro** [comment]
commits are not signed, see DCO check

**Comment by Atul-Koundal** [comment]
@yurishkuro both commits are now signed off Thanks for the review

**Review by yurishkuro — APPROVED** [review]
